src/v/cluster/tm_stm.h

```diff
@@ -439,6 +441,11 @@ class tm_stm final : public raft::persisted_stm<> {
     chunked_hash_map<kafka::transactional_id, tx_wrapper> _transactions;
 
     mutex _tx_thrashing_lock{"tm_stm::tx_thrashing_lock"};
+
+    probe _probe;
```

Comment:
not sure it qualifies, but a while back we had moved a lot of probes into unique_ptr. here was that commit with explanation https://github.com/redpanda-data/redpanda/commit/bedfce577767fcd724b0268488a00bf714cb4976 from @rockwotj

Reply:
Two things:

1. We should try to use our metrics wrappers which allow for dynamic aggregation of ntp that @StephanDollberg created in the metric library (I have not looked deep enough to see if we can).
1. If we can't use our internal metric wrappers (which are marked immovable) we should make this immovable and add a comment as to why
1. should we name this tx_probe or tx_stats? Seeing as cluster::probe is very generic.